tools/BackPopulateDeepQC.pl

```diff
@@ -0,0 +1,144 @@
+################################################################
+if (!$ENV{LORIS_CONFIG}) {
+	print STDERR "\n\tERROR: Environment variable 'LORIS_CONFIG' not set\n\n";
+	exit $NeuroDB::ExitCodes::INVALID_ENVIRONMENT_VAR;
+}
+
+if (!defined $profile || !-e "$ENV{LORIS_CONFIG}/.loris_mri/$profile") {
+    print $Help;
+    print STDERR "$Usage\n\tERROR: You must specify a valid and existing profile.\n\n";
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+
+if ( !@Settings::db ) {
+    print STDERR "\n\tERROR: You don't have a \@db setting in the file "
+                 . "$ENV{LORIS_CONFIG}/.loris_mri/$profile \n\n";
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+################################################################
+######### Establish database connection ########################
+################################################################
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+print "\nSuccessfully connected to database \n";
+
+################################################################
+######### Initialize variables #################################
+################################################################
+my $data_dir = &NeuroDB::DBI::getConfigSetting(
+                    \$dbh,'dataDirBasepath'
+                    );
+my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
+                       \$dbh,'tarchiveLibraryDir'
+                       );
+$tarchiveLibraryDir    =~ s/\/$//g;
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst)
+    =localtime(time);
+my $template = "TarLoad-$hour-$min-XXXXXX"; # for tempdir
+my $TmpDir = tempdir(
+                 $template, TMPDIR => 1, CLEANUP => 1
+             );
+my @temp     = split(/\//, $TmpDir);
+my $templog  = $temp[$#temp];
+my $LogDir   = "$data_dir/logs";
+if (!-d $LogDir) {
+    mkdir($LogDir, 0770);
+}
+my $logfile  = "$LogDir/$templog.log";
+
+################################################################
+################## Instantiate MRIProcessingUtility ############
+################################################################
+my $utility = NeuroDB::MRIProcessingUtility->new(
+                  \$dbh,$debug,$TmpDir,$logfile,
+                  $LogDir,$verbose
+              );
+
+################################################################
+# Grep tarchive list for all those entries with         ########
+# NULL in ArchiveLocationPerModality                    ########
+################################################################
+
+# Query to grep all tarchive entries
+if (!defined($TarchiveID)) {
+    $query = "SELECT TarchiveID " .
+        "FROM tarchive";
+}
+# Selecting tarchiveID is redundant here but it makes the while() loop
+# applicable to both cases; when a TarchiveID is specified or not
+else {
+    $query = "SELECT TarchiveID " .
+        "FROM tarchive ".
+        "WHERE TarchiveID = $TarchiveID ";
```

Comment:
for the queries, should use the new Moose system that Nicolas developed.

General comment: variables should not be integrated in the query, instead, use placeholders in the query and execute the query with the variables. For this example, $TarchiveID should be replaced by a placeholder in the query (`?` in Perl) and the query should be executed with the `$TarchiveID` variable: `$sth-> execute($TarchiveID)`